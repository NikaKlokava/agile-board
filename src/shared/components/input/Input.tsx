import { useFormikContext } from "formik";
import cl from "./input.module.css";

type Props = {
  formikName: string;
};

export const Input = ({ formikName }: Props) => {
  const { handleChange, setFieldValue } = useFormikContext();

  const handleDeleteCLick = (elem: any) => {
    elem.parentElement.remove();
    setFieldValue(formikName, undefined);
  };

  return (
    <div className={cl.input_container}>
      <input
        type={"text"}
        placeholder="e.g Take coffee break"
        spellCheck={false}
        className={cl.input_style}
        autoComplete="off"
        onChange={handleChange}
        name={formikName}
      />
      <div
        className={cl.delete_icon}
        onClick={(e) => handleDeleteCLick(e.target)}
      />
    </div>
  );
};
