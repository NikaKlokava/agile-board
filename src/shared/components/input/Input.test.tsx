import renderer from "react-test-renderer";
import { Input } from "./Input";

describe("Test the Input component", () => {
  test("The Input renders correctly", () => {
    const inputSnap = renderer.create(<Input withDelete={true} />).toJSON();
    expect(inputSnap).toMatchSnapshot();
  });
});
