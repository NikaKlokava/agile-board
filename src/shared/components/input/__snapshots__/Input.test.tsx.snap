// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Input component The Input renders correctly 1`] = `
<div
  className="input_container"
>
  <input
    className="input_style"
    placeholder="e.g Take coffee break"
    spellCheck={false}
    type="text"
  />
  <div
    className="delete_icon"
  />
</div>
`;
